self_test/SelfTest.ts

```diff
@@ -0,0 +1,120 @@
+  }
+
+  start () {
+    if (!this.config.disableSelfTest) {
+      this.run()
+      .catch(err => log.error(err))
+    } else {
+      log.debug('Skipping self test')
+    }
+  }
+
+  async run () {
+    const duration = 300
+    const host = this.config.publicUri
+    try {
+      const randomName = crypto.randomBytes(20).toString('hex')
+      manifestJson['manifest']['name'] = randomName
+      log.debug('manifestJson', manifestJson)
+      let response = await ilpFetch(`${host}/pods?duration=${duration}`, {
+        headers: {
+          Accept: `application/codius-v1+json`,
+          'Content-Type': 'application/json'
+        },
+        maxPrice: (this.config.hostCostPerMonth * 1000000).toString(),
+        method: 'POST',
+        body: JSON.stringify(manifestJson),
+        timeout: 70000 // 1m10s
+      })
+      log.debug('ilpFetch Resp', response)
+      if (this.checkStatus(response)) {
+        // Maybe check status in 30 seconds interval twice.
+        response = await response.json()
+        const url = new URL(this.config.publicUri)
+        await new Promise((resolve, reject) => {
+          setTimeout(() => {
+            resolve()
+          }, 10000)
+        })
+        const webSocketPromise = new Promise(resolve => {
+          const ws = new WebSocket(`wss://${response.manifestHash}.${url.host}/websockets`)
+          ws.on('open', () => {
+            log.debug('Web sockets are enabled')
+          })
+
+          ws.on('message', (message: string) => {
+            let finalMessage = JSON.parse(message)
+            if (finalMessage.websocketEnabled) {
+              resolve()
+            }
+          })
+
+          ws.on('error', (err: any) => {
+            log.error('Error on connection for websockets', err)
+            process.exit(1)
+          })
+        })
+
+        const serverPromise = new Promise(async resolve => {
+          try {
+            const serverRes = await axios.get(`https://${response.manifestHash}.${url.host}/server`)
+            const serverCheck = serverRes.data
+            log.debug('Pod upload succeeded', serverCheck)
+            if (serverCheck.imageUploaded) {
+              // resolve promise
+              resolve()
+            }
+          } catch (err) {
+            log.error('Test contract not running')
+            process.exit(1)
+          }
+        })
+        const testPromises = await Promise.all([serverPromise, webSocketPromise])
+          // Test that none of these promises are hanging for more than 60 seconds
+          // Timeout is set so that the contract has time to be pulled.
+        Promise.race([
+          testPromises,
+          new Promise((resolve, reject) => {
+            let timeout = setTimeout(function () {
+              clearTimeout(timeout)
+              reject(new Error('Could not listen to server or websocket'))
+            }, 60000)
+          })
+        ]).catch(err => {
+          log.error('Promise race err: ', err)
```

Comment:
`Promise race errr` could confuse people as a message. Something like `Error occurred when accessing self-test contract:` could be more descriptive.